src/engine/board.ts

```diff
@@ -47,4 +48,8 @@ export default class Board {
         }
         return board;
     }
+
+    public isOnBoard(square : Square) : boolean {
```

Comment:
For all places where you've done types, nitpick but its convention to do` variable: type` e.g. `square: Square` (as in no extra space after the variable name)